cli/js/testing.ts

```diff
@@ -120,70 +121,109 @@ export function test(
   TEST_REGISTRY.push(testDef);
 }
 
-interface TestStats {
-  filtered: number;
-  ignored: number;
-  measured: number;
-  passed: number;
-  failed: number;
+export interface RunTestsStartMessage {
+  kind: "runTestsStart";
+  tests: TestDefinition[];
 }
 
-export interface RunTestsOptions {
-  exitOnFail?: boolean;
-  failFast?: boolean;
-  only?: string | RegExp;
-  skip?: string | RegExp;
-  disableLog?: boolean;
-  reporter?: TestReporter;
+export interface TestStartMessage {
+  kind: "testStart";
+  test: TestDefinition;
 }
 
-enum TestStatus {
-  Passed = "passed",
-  Failed = "failed",
-  Ignored = "ignored"
-}
-
-interface TestResult {
+export interface TestEndMessage {
+  kind: "testEnd";
   name: string;
-  status: TestStatus;
+  status: "passed" | "failed" | "ignored";
   duration: number;
   error?: Error;
 }
 
-export enum TestEvent {
-  Start = "start",
-  TestStart = "testStart",
-  TestEnd = "testEnd",
-  End = "end"
+export interface RunTestsEndMessage {
+  kind: "runTestsEnd";
+  filtered: number;
+  ignored: number;
+  measured: number;
+  passed: number;
+  failed: number;
+  duration: number;
+  errors: Array<[string, Error]>;
```

Comment:
What's the reason to return `errors` and not all results?

Reply:
It's hard to decide what should go in `RunTestsEndMessage` since technically all of it is redundant. In the end I just made it contain what is commonly used in test reporters. But yeah this is an after-the-fact justification to squash `TestResult` into `TestEndMessage`.

Reply:
Reverted. https://github.com/denoland/deno/blob/458e24cef6a26e1d68e1f925e6b1348bf058f6dd/cli/js/lib.deno.ns.d.ts#L56 The `& {}` is to remove the optionality of `TestMessage["testEnd"]`.